sycl/gdb/libsycl.so-gdb.py

```diff
@@ -134,3 +139,86 @@ def match(self, class_type, method_name):
 
 
 gdb.xmethod.register_xmethod_matcher(None, AccessorOpIndexMatcher(), replace=True)
+
+### Pretty-printer implementations ###
+
+"""
+Print a cl::sycl::array
```

Comment:
What is a `cl::sycl::array`?

Reply:
It is defined in `sycl/detail/array.hpp` (see the test case) and used as a base class for several other entities.

Reply:
Except that it is not. As @bader suggested there is a `namespace detail {` there...